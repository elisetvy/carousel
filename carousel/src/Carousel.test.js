import { render, fireEvent } from "@testing-library/react";
import Carousel from "./Carousel";
import TEST_IMAGES from "./_testCommon.js";

it("works when you click on the right arrow", function() {
  const { container } = render(
    <Carousel
      photos={TEST_IMAGES}
      title="images for testing"
    />
  );
  // expect the first image to show, but not the second
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 2"]')
  ).not.toBeInTheDocument();

  // move forward in the carousel
  const rightArrow = container.querySelector(".bi-arrow-right-circle");
  fireEvent.click(rightArrow);

  // expect the second image to show, but not the first
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).not.toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 2"]')
  ).toBeInTheDocument();
});

it("works when you click on the left arrow", function() {
  const { container } = render(
    <Carousel
      photos={TEST_IMAGES}
      title="images for testing"
    />
  );
  // expect the first image to show, but not the second
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 3"]')
  ).not.toBeInTheDocument();

  // move forward in the carousel
  const rightArrow = container.querySelector(".bi-arrow-right-circle");
  fireEvent.click(rightArrow);

  // move backward in the carousel
  const leftArrow = container.querySelector(".bi-arrow-left-circle");
  fireEvent.click(leftArrow);

  // expect the first image to show, but not the second
  expect(
    container.querySelector('img[alt="testing image 2"]')
  ).not.toBeInTheDocument();
  expect(
    container.querySelector('img[alt="testing image 1"]')
  ).toBeInTheDocument();
});

it("left arrow hidden on first image", function() {
  const { container } = render(
    <Carousel
      photos={TEST_IMAGES}
      title="images for testing"
    />
  );
  const leftArrow = container.querySelector(".bi-arrow-left-circle");

  // expect left arrow to be hidden
  expect(leftArrow).not.toBeVisible();
});

it("right arrow hidden on last image", function() {
  const { container } = render(
    <Carousel
      photos={TEST_IMAGES}
      title="images for testing"
    />
  );

  const rightArrow = container.querySelector(".bi-arrow-right-circle");
  fireEvent.click(rightArrow);
  fireEvent.click(rightArrow);

  // expect right arrow to be hidden
  expect(rightArrow).not.toBeVisible();
});

it("renders without crashing", function() {
  render(<Carousel photos={TEST_IMAGES} title="Test photos"/> )
});

it("matches snapshot - initial render", function() {
  const { container } = render(<Carousel photos={TEST_IMAGES} title="Test photos"/>);
  expect(container).toMatchSnapshot();
})

it("matches snapshot - second picture", function() {
  const { container } = render(<Carousel photos={TEST_IMAGES} title="Test photos"/>);
  const rightArrow = container.querySelector(".bi-arrow-right-circle");
  fireEvent.click(rightArrow);
  expect(container).toMatchSnapshot();
})

