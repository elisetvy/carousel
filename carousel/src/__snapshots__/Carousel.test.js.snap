// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot  1`] = `
<div>
  <div
    class="Carousel"
  >
    <h1>
      Test photos
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="bi bi-arrow-left-circle"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          testing image 2
        </h4>
        <img
          alt="testing image 2"
          class="Card-image"
          src="test2.com"
        />
        <small
          class="Card-small"
        >
          Image 
          2
           of 
          3
          .
        </small>
      </div>
      <i
        class="bi bi-arrow-right-circle"
      />
    </div>
  </div>
</div>
`;

exports[`matches snapshot  2`] = `
<div>
  <div
    class="Carousel"
  >
    <h1>
      Test photos
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="bi bi-arrow-left-circle"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          testing image 2
        </h4>
        <img
          alt="testing image 2"
          class="Card-image"
          src="test2.com"
        />
        <small
          class="Card-small"
        >
          Image 
          2
           of 
          3
          .
        </small>
      </div>
      <i
        class="bi bi-arrow-right-circle"
      />
    </div>
  </div>
</div>
`;

exports[`matches snapshot -  1`] = `
<div>
  <div
    class="Carousel"
  >
    <h1>
      Test photos
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="bi bi-arrow-left-circle"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          testing image 2
        </h4>
        <img
          alt="testing image 2"
          class="Card-image"
          src="test2.com"
        />
        <small
          class="Card-small"
        >
          Image 
          2
           of 
          3
          .
        </small>
      </div>
      <i
        class="bi bi-arrow-right-circle"
      />
    </div>
  </div>
</div>
`;

exports[`matches snapshot - initial render 1`] = `
<div>
  <div
    class="Carousel"
  >
    <h1>
      Test photos
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="bi bi-arrow-left-circle"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          testing image 1
        </h4>
        <img
          alt="testing image 1"
          class="Card-image"
          src="test1.com"
        />
        <small
          class="Card-small"
        >
          Image 
          1
           of 
          3
          .
        </small>
      </div>
      <i
        class="bi bi-arrow-right-circle"
      />
    </div>
  </div>
</div>
`;

exports[`matches snapshot - second picture 1`] = `
<div>
  <div
    class="Carousel"
  >
    <h1>
      Test photos
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="bi bi-arrow-left-circle"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          testing image 2
        </h4>
        <img
          alt="testing image 2"
          class="Card-image"
          src="test2.com"
        />
        <small
          class="Card-small"
        >
          Image 
          2
           of 
          3
          .
        </small>
      </div>
      <i
        class="bi bi-arrow-right-circle"
      />
    </div>
  </div>
</div>
`;

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Carousel"
  >
    <h1>
      Test photos
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="bi bi-arrow-left-circle"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          testing image 1
        </h4>
        <img
          alt="testing image 1"
          class="Card-image"
          src="test1.com"
        />
        <small
          class="Card-small"
        >
          Image 
          1
           of 
          3
          .
        </small>
      </div>
      <i
        class="bi bi-arrow-right-circle"
      />
    </div>
  </div>
</div>
`;

exports[`matches snapshot 2`] = `
<div>
  <div
    class="Carousel"
  >
    <h1>
      Test photos
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="bi bi-arrow-left-circle"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          testing image 2
        </h4>
        <img
          alt="testing image 2"
          class="Card-image"
          src="test2.com"
        />
        <small
          class="Card-small"
        >
          Image 
          2
           of 
          3
          .
        </small>
      </div>
      <i
        class="bi bi-arrow-right-circle"
      />
    </div>
  </div>
</div>
`;

exports[`matches snapshot w 1`] = `
<div>
  <div
    class="Carousel"
  >
    <h1>
      Test photos
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="bi bi-arrow-left-circle"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          testing image 1
        </h4>
        <img
          alt="testing image 1"
          class="Card-image"
          src="test1.com"
        />
        <small
          class="Card-small"
        >
          Image 
          1
           of 
          3
          .
        </small>
      </div>
      <i
        class="bi bi-arrow-right-circle"
      />
    </div>
  </div>
</div>
`;

exports[`matches snapshot wi 1`] = `
<div>
  <div
    class="Carousel"
  >
    <h1>
      Test photos
    </h1>
    <div
      class="Carousel-main"
    >
      <i
        class="bi bi-arrow-left-circle"
      />
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          testing image 1
        </h4>
        <img
          alt="testing image 1"
          class="Card-image"
          src="test1.com"
        />
        <small
          class="Card-small"
        >
          Image 
          1
           of 
          3
          .
        </small>
      </div>
      <i
        class="bi bi-arrow-right-circle"
      />
    </div>
  </div>
</div>
`;
